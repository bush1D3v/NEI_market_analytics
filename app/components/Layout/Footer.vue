<script setup lang="ts">
import Image from "@/tags/Image.vue";
import RouterLink from "@/tags/RouterLink.vue";
import ToggleTheme from "@/components/ToggleTheme.vue";
import Link from "@/tags/Link.vue";
import { t } from "i18next";
import { FooterDto } from "@/components/Dto/Layout/FooterDto";
import { useRoute } from "vue-router";
import { GithubLogoIcon } from "@radix-icons/vue";

const route = useRoute();
</script>

<template>
    <footer class="layout fixed bottom-0 xs:relative xs:container justify-center min-w-full border-t text-dark"
        data-testid="Footer">
        <ul class="hidden xs:flex gap-0 flex-row justify-around items-center w-full py-12">
            <li class="flex flex-col gap-4">
                <Image src="/logo.svg" alt="NEI Logo" width="200" height="71.59" />
                <span>© 2024 Todos os direitos reservados.</span>
            </li>
            <li class="flex flex-col gap-5">
                <h3>Info</h3>
                <ul class="flex flex-col gap-2">
                    <li>
                        <RouterLink to="/about-us" class="hover:opacity-50">
                            <span>Sobre Nós</span>
                        </RouterLink>
                    </li>
                    <li>
                        <RouterLink to="/privacy-policy" class="hover:opacity-50">
                            <span>Política de Privacidade</span>
                        </RouterLink>
                    </li>
                    <li>
                        <RouterLink to="/cookie-policy" class="hover:opacity-50">
                            <span>Política de Cookies</span>
                        </RouterLink>
                    </li>
                </ul>
            </li>
            <li class="flex flex-col gap-5">
                <h3>Modo Escuro</h3>
                <ToggleTheme variant="airplane" />
                <h3>Redes Sociais</h3>
                <Link href="https://github.com/bush1D3v/NEI_market_analytics/tree/dev" target="_blank">
                <div class="flex gap-3">
                    <GithubLogoIcon class="w-6 h-6" />
                    <span>Ver no GitHub</span>
                </div>
                </Link>
            </li>
        </ul>
        <ul class="grid grid-cols-5 bg-secondaryDarker w-full h-full xs:hidden">
            <li v-for="(data, i) in FooterDto" :key="i"
                :class="{ 'text-primary': route.path === data.path, 'cursor-pointer hover:text-primary duration-150 ease-in-out p-3': true }">
                <RouterLink :to="data.path" class="flex flex-col items-center">
                    <component :is="data.icon" class="w-6 h-6" />
                    <h5>{{ t(data.title) }}</h5>
                </RouterLink>
            </li>
        </ul>
    </footer>
</template>

<style scoped>
h3 {
    @apply text-text;
}
</style>
