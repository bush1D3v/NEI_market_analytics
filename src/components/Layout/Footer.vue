<script setup lang="ts"></script>

<template>
    <footer class="layout container justify-center min-w-full border-t" data-testid="Footer">
        Footer
    </footer>
</template>
