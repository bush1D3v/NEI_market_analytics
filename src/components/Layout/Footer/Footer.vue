<script setup lang="ts"></script>

<template>
    <footer class="layout container min-w-full">
        Footer
    </footer>
</template>
