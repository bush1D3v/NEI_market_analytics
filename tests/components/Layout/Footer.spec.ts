import {test, expect, describe} from "vitest";
import Footer from "@/components/Layout/Footer.vue";
import {mount} from "@vue/test-utils";

describe("Footer Component Tests", () => {
	const wrapper = mount(Footer);
	const footerComponent = wrapper.findComponent(Footer);

	test("Should be able to render the component correctly", async () => {
		expect(footerComponent).toBeDefined();
	});

	test("Should be able to snapshot the component correctly", async () => {
		expect(footerComponent.html()).toMatchSnapshot();
	});

	test("Should be able to render the component with the correct text", async () => {
		expect(footerComponent.text()).toStrictEqual("Footer");
	});

	test("Should be able to render the component with the correct classes", async () => {
		expect(footerComponent.classes()).toStrictEqual([
			"layout",
			"container",
			"justify-center",
			"min-w-full",
			"border-t",
		]);
	});
});
