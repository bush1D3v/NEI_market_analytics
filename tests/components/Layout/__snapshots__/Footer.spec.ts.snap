// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer Component Tests > Should be able to snapshot the component correctly 1`] = `
"<footer data-v-936543d0="" class="layout container justify-center min-w-full border-t text-dark" data-testid="Footer">
  <ul data-v-936543d0="" class="flex justify-around w-full py-12">
    <li data-v-936543d0="" class="flex flex-col gap-4">
      <image-stub data-v-936543d0="" src="/logo.svg" alt="NEI Logo" width="200" height="71.59"></image-stub><span data-v-936543d0="">Â© 2024 Todos os direitos reservados.</span>
    </li>
    <li data-v-936543d0="" class="flex flex-col gap-5">
      <h3 data-v-936543d0="">Info</h3>
      <ul data-v-936543d0="" class="flex flex-col gap-2">
        <li data-v-936543d0="">
          <router-link-stub data-v-936543d0="" to="/about-us" class="hover:opacity-50"></router-link-stub>
        </li>
        <li data-v-936543d0="">
          <router-link-stub data-v-936543d0="" to="/privacy-policy" class="hover:opacity-50"></router-link-stub>
        </li>
        <li data-v-936543d0="">
          <router-link-stub data-v-936543d0="" to="/cookie-policy" class="hover:opacity-50"></router-link-stub>
        </li>
      </ul>
    </li>
    <li data-v-936543d0="" class="flex flex-col gap-5">
      <h3 data-v-936543d0="">Modo Escuro</h3>
      <toggle-theme-stub data-v-936543d0="" variant="airplane"></toggle-theme-stub>
      <h3 data-v-936543d0="">Redes Sociais</h3>
      <link-stub data-v-936543d0="" href="https://github.com/bush1D3v/NEI_market_analytics/tree/dev" target="_blank"></link-stub>
    </li>
  </ul>
</footer>"
`;
